src/prefect/settings.py

```diff
@@ -368,6 +368,36 @@ def check_for_deprecated_cloud_url(settings, value):
     return deprecated_value or value
 
 
+def check_for_misconfigured_api_url(api_url):
```

Comment:
I'm wondering if there's a more ideal place than `settings.py` for this

Reply:
I think `settings.py` is appropriate, but maybe implemented as a validator when calling `PREFECT_API_URL.value` since there are multiple ways of setting the variable, but only one way to retrieve it?

Reply:
Indeed, the validator will check the existing profile values, whereas during set() we don't look at current values but rather the ones attempting to be set.